Modules/_csv.c

```diff
@@ -1466,9 +1502,10 @@ csv_field_size_limit(PyObject *module, PyObject *args)
                          "limit must be an integer");
             return NULL;
         }
-        _csvstate_global->field_limit = PyLong_AsLong(new_limit);
-        if (_csvstate_global->field_limit == -1 && PyErr_Occurred()) {
-            _csvstate_global->field_limit = old_limit;
+       
+        state->field_limit = PyLong_AsLong(new_limit);
+        if (state->field_limit == -1 && PyErr_Occurred()) {
+            state->field_limit = old_limit;
```

Comment:
Would it be possible to store PyLong_AsLong() in a temporary object, and only update field_limit on success? It would avoid to have to save/restore field_limit value.